tests/core/test_api.py

```diff
@@ -49,11 +51,23 @@ def test_generate_resource_urls_with_id(test_config: FidesctlConfig) -> None:
 
 # Integration Tests
 @pytest.mark.integration
-def test_api_ping(test_config: FidesctlConfig) -> None:
-    assert (
-        _api.ping(test_config.cli.server_url + API_PREFIX + "/health").status_code
-        == 200
-    )
+@pytest.mark.parametrize(
+    "database_health, expected_status_code",
+    [("healthy", 200), ("needs migration", 200), ("unhealthy", 503)],
+)
+def test_api_ping(
+    test_config: FidesctlConfig,
+    database_health: str,
+    expected_status_code: int,
+    monkeypatch: MonkeyPatch,
+) -> None:
+    def mock_get_db_health() -> str:
+        return database_health
+
+    # not working :(
```

Comment:
It works if you call the health route directly through the `starlette` test client. The mock function also needs `*args` and `**kwargs`.

```py
from starlette.testclient import TestClient

@pytest.fixture()
def client():
    with TestClient(main.app) as test_client:
        yield test_client


# Integration Tests
@pytest.mark.integration
@pytest.mark.parametrize(
    "database_health, expected_status_code",
    [("healthy", 200), ("needs migration", 200), ("unhealthy", 503)],
)
def test_api_ping(
    test_config: FidesctlConfig,
    database_health: str,
    expected_status_code: int,
    monkeypatch: MonkeyPatch,
    client,
) -> None:
    def mock_get_db_health(*args, **kwargs) -> str:
        return database_health

    monkeypatch.setattr(main, "get_db_health", mock_get_db_health)
    response = client.get(test_config.cli.server_url + API_PREFIX + "/health")

    assert response.status_code == expected_status_code
```

Why it doesn't work with calling through the `ping` I can't figure out. I think it has something to do with where the `get_db_health` function needs to get patched, but I can't find anywhere to patch it to make it work 🤷